src/test/java/io/github/thebusybiscuit/slimefun4/testing/tests/items/TestDamageableItem.java

```diff
@@ -0,0 +1,126 @@
+        item.register(plugin);
+        return item;
+    }
+
+    @Test
+    @DisplayName("Test DamageableItem actually damages")
+    void testDamageableItemDamagesItem() {
+        DamageableMock unDamageableItem = getDummyItem("UDM", false, null, null);
+        // Check if damageable is successfully registered
+        Assertions.assertFalse(unDamageableItem.isDamageable());
+
+        Player player = server.addPlayer();
+        ItemStack unDamageableDummy = unDamageableItem.getItem().clone();
+        unDamageableItem.damageItem(player, unDamageableDummy);
+        Assertions.assertTrue(unDamageableDummy.hasItemMeta());
+        // Check if the item is not damaged as we intended it to not be
+        Assertions.assertEquals(((Damageable) unDamageableDummy.getItemMeta()).getDamage(), 0);
+
+        DamageableMock damageableItem = getDummyItem("DM", true, null, null);
+        // Check if damageable is successfully registered
+        Assertions.assertTrue(damageableItem.isDamageable());
+
+        ItemStack damageableDummy = damageableItem.getItem().clone();
+        damageableItem.damageItem(player, damageableDummy);
+        Assertions.assertTrue(damageableDummy.hasItemMeta());
+        // Check if the item is damaged as we intended it to not be
+        Assertions.assertEquals(((Damageable) damageableDummy.getItemMeta()).getDamage(), 1);
+    }
+
+    @Test
+    @DisplayName("Test if DamageableItem cares about unbreaking levels")
+    void testDamageableItemCaresUnbreaking() {
+        DamageableMock noUnbreakingItem = getDummyItem("NU", true, null, null);
+        DamageableMock iiiUnbreakingItem = getDummyItem("IIIU", true, Enchantment.DURABILITY, 3);
+        DamageableMock xUnbreakingItem = getDummyItem("XU", true, Enchantment.DURABILITY, 10);
+        ItemStack noUnbreakingItemIS = noUnbreakingItem.getItem().clone();
+        ItemStack iiiUnbreakingItemIS = iiiUnbreakingItem.getItem().clone();
+        ItemStack xUnbreakingItemIS = xUnbreakingItem.getItem().clone();
+        Player player = server.addPlayer();
+        for (int i = 0; i < 500; ++i) {
+            noUnbreakingItem.damageItem(player, noUnbreakingItemIS);
+            noUnbreakingItem.damageItem(player, iiiUnbreakingItemIS);
+            noUnbreakingItem.damageItem(player, xUnbreakingItemIS);
+        }
+        Assertions.assertTrue(noUnbreakingItemIS.hasItemMeta());
+        Assertions.assertTrue(iiiUnbreakingItemIS.hasItemMeta());
+        Assertions.assertTrue(xUnbreakingItemIS.hasItemMeta());
+        Assertions.assertTrue(((Damageable) xUnbreakingItemIS.getItemMeta()).getDamage() < ((Damageable) iiiUnbreakingItemIS.getItemMeta()).getDamage());
+        Assertions.assertTrue(((Damageable) iiiUnbreakingItemIS.getItemMeta()).getDamage() < ((Damageable) noUnbreakingItemIS.getItemMeta()).getDamage());
+    }
+
+    private static class DamageableMock extends SlimefunItem implements DamageableItem {
```

Comment:
Do not declare classes inside of other classes. Make a seperate file for each class.
Check the CONTRIBUTING.md for a reference to our style guidelines.